presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1245,6 +1245,7 @@ private void prepareAlterPartition(MetastoreContext metastoreContext, HdfsContex
                         TRANSACTION_CONFLICT,
                         format("The partition that this transaction modified was deleted in another transaction. %s %s", partition.getTableName(), partition.getValues()));
             }
+            partition.setCreateTime(oldPartition.get().getCreateTime());
```

Comment:
Why do we need to do this here ? can we set it as the partition creation time itself ?

Reply:
The partition has been judged as a kind of 'alter' action here. We only need to add one row code to replace the new partition createTime with the old one.
If we set it at the partition creation time itself, we need to visit hivemetastore one more time to get old partition. (Although hive metastore has cache)

Reply:
```
            partition = Partition.builder(partition)
                    .setCreateTime(oldPartition.get().getCreateTime())
                    .build();
```

or may be move this into a new function ?